src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends T> type) {
+        List<Machine> machines = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
+            machines.addAll(bulldozerProducer.get());
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+            machines.addAll(excavatorProducer.get());
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> truckProducer = new TruckProducer();
+            machines.addAll(truckProducer.get());
+        }
+        return machines;
```

Comment:
Instead of returning a specific type List or null in `getAll()` when no producer has been found, return an empty List. This avoids the need for additional null checks and potential NullPointerExceptions.